Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1887,17 +1898,48 @@
     CGAffineTransform userTransform =
         CGAffineTransformConcat(curState->curTransform, CGAffineTransformMake(1.0f / _scale, 0, 0, -1.0f / _scale, 0, height / _scale));
 
-    // Apply the text position in user space, not in text space
-    userTransform = CGAffineTransformTranslate(userTransform, curState->curTextPosition.x, curState->curTextPosition.y);
+    CGAffineTransform finalTransform = CGAffineTransformConcat(textTransform, userTransform);
 
-    // Apply the two transforms giving us the final result
-    CGAffineTransform transform = CGAffineTransformConcat(textTransform, userTransform);
-    imgRenderTarget->SetTransform(D2D1::Matrix3x2F(transform.a, transform.b, transform.c, transform.d, transform.tx, transform.ty));
+    if ((fabs(finalTransform.a * glyphRun->fontEmSize) <= c_glyphThreshold &&
```

Comment:
ugh i hate this a lot but i guess there's not really a better way for the time being. minor optimization thought since i'm in that headspace: check d before c, since d is less likely to evaluate to true than c is, as c is often 0.